__tests__/unit/http_adapter.test.ts

```diff
@@ -239,6 +240,42 @@ describe('HttpAdapter', () => {
     }
   })
 
+  describe('User-Agent', () => {
+    class MyTestHttpAdapter extends BaseHttpAdapter {
+      constructor(application_id?: string) {
+        super(
+          { application_id } as ConnectionParams,
+          new TestLogger(),
+          {} as Http.Agent
+        )
+      }
+      protected createClientRequest(): Http.ClientRequest {
+        return {} as any
+      }
+      public getDefaultHeaders() {
+        return this.headers
+      }
+    }
+
+    it('should have proper user agent without app id', async () => {
+      const myHttpAdapter = new MyTestHttpAdapter()
+      const headers = myHttpAdapter.getDefaultHeaders()
+      expect(headers['Authorization']).toMatch(/^Basic [A-Za-z0-9/+=]+$/)
+      expect(headers['User-Agent']).toMatch(
+        /^clickhouse-js\/[0-9\\.]+? \(lv:nodejs\/v[0-9\\.]+?; os:(?:linux|darwin|win32)\)$/
```

Comment:
optional: I'd remove the assertion against `linux|darwin|win32` to prevent flakiness. We can match a non-empty string instead.

Reply:
I doubt that it will be flaky.
https://nodejs.org/api/os.html#os_os_platform
> Possible values are 'aix', 'darwin', 'freebsd','linux', 'openbsd', 'sunos', and 'win32'.

That's very unlikely that the code will be run on `aix`, `openbsd`, `freebsd` or `sunos`